src/dataSources/JSON.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+var Null = require('./Null');
+
+/**
+ * @constructor
+ */
+var JSON = Null.extend('JSON',  {
```

Comment:
Using .extend here also makes this object extensible. Are we sure we want that? I know it was extensible in hyper-analytics, but in practice it was never extended from. Let's just make it a regular function with a prototype.